src/search.cpp

```
@@ -113,14 +113,15 @@ void update_correction_history(const Position& pos,
-    const int    bonus4 = (bonus * 63 / 128) * mask;
-    (*(ss - 2)->continuationCorrectionHistory)[pc][to] << bonus2;
-    (*(ss - 4)->continuationCorrectionHistory)[pc][to] << bonus4;
+    if (m.is_ok())
+    {
+        const Square to = m.to_sq();
+        const Piece  pc = pos.piece_on(to);
+        const int    b2 = bonus * 126 / 128;
+        const int    b4 = bonus * 63 / 128;
+        SharedHistories::contcorr_update(*(ss - 2)->continuationCorrectionHistory, pc, to, b2);
+        SharedHistories::contcorr_update(*(ss - 4)->continuationCorrectionHistory, pc, to, b4);
+    }
```

Cppcheck (2.20.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)